### ISSUE TITLE
vr support?

### ISSUE THREAD

**Issue by xAstroBoy** [lossless1024/StreaMonitor#11]
stripchat \ xhamsterlivesome models have vr cams, it would  be nice if you add the vr stream as well, is a wss link.

**Comment by lossless1024** [comment]
This is a really good suggestion. It might require a change in the framework, but worth the effort, so please be patient, this will be added.

**Comment by xAstroBoy** [comment]
> I started to implement this, and works pretty good, but there are a few hard decisions that should be made, so I need a bit more time. And also I will have to add some post-processing, because the timestamps are aligned to the beginning of the stream and it is kinda inconvenient when the file starts from x hours.

as long we get the raw video without any conversion done by the browser, as the whole purpose is to replay it in a VR headset

**Comment by lossless1024** [comment]
It's done. If you pull to the latest version, you can check. To add VR channel use `scvr` as site. Currently this is implemented as different sites so you'll have to add Stripchat and StripchatVR channels separately.

**Comment by lossless1024** [comment]
The post-processing is still missing, I'll fix it ASAP

**Comment by lossless1024** [comment]
It is done with the latest commit, please check

**Comment by xAstroBoy** [comment]
@lossless1024
Found a better way to fix timestamps without postprocessing

 ffmpeg -ignore_editlist 1 -i tmp.mp4 -codec copy output.mp4

use this command

**Comment by lossless1024** [comment]
The code does exactly this, only without the ignore_editlist flag

**Comment by xAstroBoy** [comment]
> The code does exactly this, only without the ignore_editlist flag

ignore editlist makes it faster as well, i tested and it fixed the entire video, probably because it skips the missing files clips from the stream and makes ffmpeg regenerate the timestamps

**Comment by lossless1024** [comment]
Okay, then I add it.

**Comment by lossless1024** [comment]
I pushed the modification. Also implemented some looping to avoid new files on every reconnection.

**Comment by xAstroBoy** [comment]
> I pushed the modification. Also implemented some looping to avoid new files on every reconnection. Try it please.

still happens when a VR stream drops.

**Comment by lossless1024** [comment]
It shouldn't if the bot can continue immediately. However if it couldn't, then a new file will be created. Sorry, I can't make it better right now. If you can, feel free to submit a PR.

**Comment by xAstroBoy** [comment]
@lossless1024 i wonder, is possible to redirect the video thru a local url, then ffmpeg does the rest? this way we have a proper video and there's no need for post processing.

**Comment by lossless1024** [comment]
TL;DR No.

Every websocket streaming implementation is custom as I saw. AFAIK FFmpeg doesn't even support websockets. This is the reason why I didn't want to include any websocket magic at all. Unfortunately there are these sites which use it. However I totally agree with the fact that websockets are pretty good for low-latency uses. 
On the actual VR site you don't see, but the connection drops there too, but you don't notice, because there are buffers and also the stream restarts almost immediately. The latest fix tries to continue the stream, but if it can't immediately, it restarts itself. Also the regular hls streams drop sometimes, you have to accept this for now. You can concat the videos if you want.

**Comment by xAstroBoy** [comment]
> TL;DR No.
> 
> Every websocket streaming implementation is custom as I saw. AFAIK FFmpeg doesn't even support websockets. This is the reason why I didn't want to include any websocket magic at all. Unfortunately there are these sites which use it. However I totally agree with the fact that websockets are pretty good for low-latency uses. On the actual VR site you don't see, but the connection drops there too, but you don't notice, because there are buffers and also the stream restarts almost immediately. The latest fix tries to continue the stream, but if it can't immediately, it restarts itself. Also the regular hls streams drop sometimes, you have to accept this for now. You can concat the videos if you want.

im saying, emulate a local videoplayer that ffmpeg could intercept, and there download the video.

**Comment by lossless1024** [comment]
Yes, kinda possible, but not that way! That is way too complicated and inefficient. And you probably won't have any benefits. The downloader does the downloading, and the ffmpeg process would also stop when an interruption happens, so it would produce the same splitted files.
TBH, simply retrying a few times would maybe improve it, but I don't have time for it right now.

**Comment by xAstroBoy** [comment]
![image](https://user-images.githubusercontent.com/92211398/196563006-312f8e43-10ff-4f52-af02-49d6b0c1b086.png)
@lossless1024 with one of my favourite models.

**Comment by lossless1024** [comment]
Pushed a fix
